### PR TITLE
Add test for NaNs in metadata

### PR DESCRIPTION
**Pull request by gibsramen**
Relevant to #128 

Added NaN to both sample & feature metadata and perform similar test to `to_dict`. Also calls `make_empress` per @ElDeveloper's suggestion.

### CODE REVIEW

**Comment by gibsramen** [comment]
Didn't realize there was an update to `test_core.py`. Will address.

**Comment by gibsramen** [comment]
So it looks like with #228 some of the handling of NaNs may have changed. When I wrote ac9d97f it was sufficient to set `dict_a_nan["sample_data"]["Sample2"]["Metadata4"]` to `np.nan`, whereas now the equivalent requires comparing to the *string* `nan` (i.e. `dict_a_nan["compressed_sample_metadata"][1][3] = str(np.nan)`).

Just want to confirm that this is desired behavior/handling.

File: tests/python/test_core.py [hunk 1]
```diff
@@ -233,6 +233,37 @@ def test_to_dict_with_feature_metadata(self):
...
+
+        tree_data = obs['tree_data']
+        exp = dict_a_nan['tree_data']
+        self.assert_almost_equal_tree_data(tree_data, exp)
+
+        dict_a_nan.pop('tree_data')
+        obs.pop('tree_data')
+        self.assertEqual(obs, dict_a_nan)
+
+        viz.make_empress()
```
Comment (ElDeveloper) [inline tests/python/test_core.py:265]: I suggest adding some minor checks that the template rendered correctly.

```suggestion
        res = viz.make_empress()
        
        self.assertTrue('empressRequire' in res)
        self.assertTrue('empress = new Empress' in res)
        self.assertTrue('emperor_require_logic' not in res)
```

File: tests/python/test_core.py [hunk 2]
```diff
@@ -233,6 +233,37 @@ def test_to_dict_with_feature_metadata(self):
...
+        nan_feature_metadata.at["a", "fmdcol2"] = np.nan
+
+        viz = Empress(self.tree, self.table, nan_sample_metadata,
+                      nan_feature_metadata,
+                      filter_unobserved_features_from_phylogeny=False)
+        obs = viz._to_dict()
+        dict_a_nan = copy.deepcopy(DICT_A)
+        dict_a_nan["sample_data"]["Sample2"]["Metadata4"] = np.nan
+        dict_a_nan["tip_metadata"] = {
+            "a": {"fmdcol1": "asdf", "fmdcol2": np.nan}
```
Comment (ElDeveloper) [inline tests/python/test_core.py:250]: I don't believe np.nan objects will serialize correctly once run through the JINJA template. I think these should be strings i.e. "nan".

**Comment by ElDeveloper** [comment]
@gibsramen thanks for the PR, and welcome 🎉! You are correct we need to check for these to be strings, not np.nan objects. Otherwise JINJA's tojson is going to fail.

**Comment by ElDeveloper** [comment]
Thanks @gibsramen, and congrats on your first contribution to Empress! 🌲 ✨

**Comment by kwcantrell** [comment]
Thanks @gibsramen!

**Pull request merged**